fv3core/testing/validation.py

```diff
@@ -52,15 +62,15 @@ def subset_output(self, varname: str, output: np.ndarray) -> np.ndarray:
             like to validate against reference data
             """
             if varname in self._selective_argument_names:
-                output = output[self._validation_slice]
+                output = output[self._validation_slice[varname]]
             return output
 
         def _set_nans(self, kwargs):
             for name in set(kwargs.keys()).intersection(self._selective_argument_names):
                 array = kwargs[name]
-                validation_data = np.copy(array[self._validation_slice])
+                validation_data = np.copy(array[self._validation_slice[name]])
                 array[:] = np.nan
-                array[self._validation_slice] = validation_data
+                array[self._validation_slice[name]] = validation_data
```

Comment:
If you use the following for this block and the suggestion for subset_output, you can delete `self._selective_argument_names` altogether:
```
            for name, validation_slice in self._validation_slice.items():
                if name in kwargs:
                    array = kwargs[name]
                    validation_data = np.copy(array[validation_slice])
                    array[:] = np.nan
                    array[validation_slice] = validation_data
```